projects/ad7746-ebz/src/app/headless.c

```diff
@@ -0,0 +1,170 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include <drivers/pwr/adi_pwr.h>
+
+#include "ad7746.h"
+#include "i2c.h"
+#include "delay.h"
+#include "print_log.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "uart_stdio.h"
+
+
+int32_t platform_init()
+{
+	if (ADI_PWR_SUCCESS != adi_pwr_Init())
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_HCLK, 1u))
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_PCLK, 1u))
+		return FAILURE;
+
+	adi_initComponents();
+	return 0;
+}
+
+int main(void)
+{
+	int32_t ret;
+	char input[10];
+	uint32_t capData = 0;
+	uint32_t min = 0;
+	uint32_t max = 0;
+	uint32_t mm = 0;
+	uint32_t temperature = 0;
+	struct ad7746_init_param adcip;
+	struct ad7746_dev *adc = NULL;
+
+	struct uart_desc *uart;
+	struct aducm_uart_init_param xuip = {
+		.parity = UART_NO_PARITY,
+		.stop_bits = UART_ONE_STOPBIT,
+		.word_length = UART_WORDLEN_8BITS
+	};
+	struct uart_init_param uip = {
+		.device_id = 0,
+		.baud_rate = 115200,
+		.size = UART_CS_8,
+		.parity = UART_PAR_NO,
+		.stop = UART_STOP_1,
+		.extra = &xuip,
+	};
+	struct i2c_platform_ops ops = {
```

Comment:
Ops are not needed.
They are needed if https://github.com/analogdevicesinc/no-OS/blob/master/drivers/i2c/i2c.c is compiled.
If this will be the case, then this should be defined here https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/aducm3029/i2c.c and the function renamed or declared static.

Reply:
Well, I guess this shows that i2c stuff on aducm needs a refactoring - either use ops, or don't use ops. There shouldn't be possible to set ops if they are not needed. Meanwhile I'll remove them from the the app init.

Reply:
Added to aducm todo list :)